diff --git a/Lib/test/test_grammar.py b/Lib/test/test_grammar.py
--- a/Lib/test/test_grammar.py
+++ b/Lib/test/test_grammar.py
@@ -77,8 +77,8 @@
 
     def test_float_exponent_tokenization(self):
         # See issue 21642.
-        self.assertEqual(1 if 1else 0, 1)
-        self.assertEqual(1 if 0else 0, 0)
+        self.assertEqual(1 if 1 else 0, 1)
+        self.assertEqual(1 if 0 else 0, 0)
         self.assertRaises(SyntaxError, eval, "0 if 1Else 0")
 
     def testStringLiterals(self):
